### PR TITLE
feat: numpy loadtxt frontend function

### LINKED ISSUE #23417: loadtxt

### PR DESCRIPTION
**Pull request by khawajaJunaid**
# PR Description
This PR adds the loadtxt frontend functionality for `numpy.loadtxt`

## Related Issue

Close #23417 

## Checklist

- [x] Did you add a function?
- [x] Did you add the tests?
- [ ] Did you run your tests and are your tests passing?
- [x] Did pre-commit not fail on any check?
- [x] Did you follow the steps we provided?

### CODE REVIEW

**Comment by khawajaJunaid** [comment]
hi @YushaArif99 can you guide me on how I can implement the test function correctly

**Comment by khawajaJunaid** [comment]
@YushaArif99 can you please review so I can make necessary fixes

**Review by YushaArif99 — CHANGES_REQUESTED** [review]
Hi @khawajaJunaid 

Apologies for the delay in review. Have been quite swamped with work recently and so haven't had the time to review any PRs 😓 

With regards to your PR, I am a bit confused as to why you've gone on to implementing the `loadtxt` function for **all** backends. 

I think the logic seems generic enough that you could implement it as a compositional function in `ivy` rather than having backend-specific implementations. You might've already had a look into this but in case you haven't, try to read through the implementation of [`np.loadtxt`](https://github.com/numpy/numpy/blob/v1.26.0/numpy/lib/npyio.py#L1100) and see if we can reuse the logic purely using ivy functions.

On a side note, I also think there's a possibility of directly calling `np.loadtxt` in the frontend and then converting the np ndarray to the correct type (i.e `np ndarray` -> `ivy array` -> `np frontend array`). This will add a dependency to Numpy but I think that is the assumption that `ivy` works with as well.

**Comment by khawajaJunaid** [comment]
hello @YushaArif99, thanks for the reply :) ! 
My first approach after reading the documentation was that the numpy backend had the loadtxt function so i decided to go with the mixed functon implementation i.e that you see implemented in different backends. However, I went through the numpy.loadtxt function source code and found that It can be implemented in ivy and thats what I have been doing (havent pushed the commits yet). I have tried to reuse the np.loadtxt function, and wrote it in ivy creation helpers so i can implement it as a composite function, there are some functions within the loadtxt that use various numpy core libraries that has been the issue for me

**Comment by khawajaJunaid** [comment]
@YushaArif99 can I get your discord for quick questions, I raised the issue in the frontends > numpy and **felix** said that we should not add numpy as a dependency (want to discuss this as well). I wanted your discord to discuss the implementation, there are a lot of arguments and functionalities that numpy.loadtxt provides which also use numpy's core libraries coded in C

**Review by YushaArif99 — COMMENTED** [review]
Hi @khawajaJunaid 

Sure. Here's my discord username : `yushkk99`

**Comment by khawajaJunaid** [comment]
Hello @YushaArif99,
Thanks I have reached out to you on discord

**Comment by Sarvesh-Kesharwani** [comment]
Hi @khawajaJunaid , 
please resolve the merge conflicts so we can proceed with your pull request.

File: ivy/functional/backends/numpy/creation.py [hunk 1]
```diff
@@ -312,4 +312,36 @@ def triu_indices(
     *,
     device: Optional[str] = None,
 ) -> Tuple[np.ndarray]:
-    return tuple(np.asarray(np.triu_indices(n=n_rows, k=k, m=n_cols)))
+    return tuple(
+        _to_device(np.asarray(np.triu_indices(n=n_rows, k=k, m=n_cols)), device=device)
+    )
```
Comment (Sarvesh-Kesharwani) [inline ivy/functional/backends/numpy/creation.py:317]: Please remove these changes unrelated to your PR.

**Comment by Sarvesh-Kesharwani** [comment]
Hi @khawajaJunaid,
image
image
Please work on fixing these errors and other suggested changes.

**Review by Sarvesh-Kesharwani — APPROVED** [review]

**Pull request closed without merging**